core/src/main/scala/spinal/core/Units.scala

```diff
@@ -80,7 +80,15 @@ abstract class PhysicalNumber[T <: PhysicalNumber[_]](protected val value: BigDe
   def /(that: BigDecimal) = newInstance(this.value / that)
   def %(that: BigDecimal) = newInstance(this.value % that)
 
-  def max(that: T) = newInstance(value.max(that.value))
+  def ==(that: T): Boolean = this.value == that.value
```

Comment:
```suggestion
  def ==(that: T): Boolean = this.value == that.value
  def !=(that: T): Boolean = this.value != that.value
```

Hmm maybe `!=` too? Or scala implements it automatically from `==`?

Also I'm not sure of what happens when using the `==` method above, can unit tests be added, please?

I find references defining `canEqual`, `equals` and `hashCode`, maybe that should be done instead?

Reply:
Thanks, I actually wanted to remove `==` as well for symmetry, and `!=` was removed for the breakage it caused in the GHDL backend, see #998.
`==` is actually the same as `equals`